src/Hosting/Server.IntegrationTesting/src/Deployers/SelfHostDeployer.cs

```diff
@@ -165,7 +184,15 @@ protected async Task<(Uri url, CancellationToken hostExitToken)> StartSelfHostAs
                     Logger.LogInformation("host process ID {pid} shut down", HostProcess.Id);
 
                     // If TrySetResult was called above, this will just silently fail to set the new state, which is what we want
-                    started.TrySetException(new Exception($"Command exited unexpectedly with exit code: {HostProcess.ExitCode}"));
+                    if (allowRetry)
+                    {
+                        started.TrySetException(new InvalidOperationException("Failed to bind to port in HttpSys."));
```

Comment:
Why flow this via an exception at all? Changing the return parameters would be clearer.